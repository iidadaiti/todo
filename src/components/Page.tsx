import '@spectrum-css/page'
import React, { ComponentProps, FC, PropsWithChildren } from 'react'
import { mergeClassNames } from '../utils/classNames'

export interface PageProps extends Omit<ComponentProps<'div'>, 'children'> {
  theme?: 'light'
  scale?: 'medium' | 'large'
}

export const Page: FC<PropsWithChildren<PageProps>> = ({
  theme = 'light',
  scale = 'medium',
  className,
  ...props
}) => {
  return (
    <div
      className={mergeClassNames(
        'spectrum',
        `spectrum--${scale}`,
        `spectrum--${theme}`,
        className
      )}
      {...props}
    ></div>
  )
}
