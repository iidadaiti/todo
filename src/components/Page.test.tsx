import { render } from '@testing-library/react'
import React from 'react'
import { Page } from './Page'

test('Page snapshot', () => {
  const result = render(<Page></Page>)
  expect(result.asFragment()).toMatchSnapshot()
})
