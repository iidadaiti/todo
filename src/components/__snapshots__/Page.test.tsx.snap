// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="spectrum spectrum--medium spectrum--light"
  />
</DocumentFragment>
`;
